lms/djangoapps/grades/tests/test_models.py

```diff
@@ -183,29 +183,39 @@ def test_update_grade(self):
         """
         Tests model update, and confirms error when updating a nonexistent model.
         """
-        with self.assertRaises(PersistentSubsectionGrade.DoesNotExist):
-            PersistentSubsectionGrade.update_grade(**self.params)
-        PersistentSubsectionGrade.objects.create(**self.params)
-        self.params['earned_all'] = 12
-        self.params['earned_graded'] = 8
-        PersistentSubsectionGrade.update_grade(**self.params)
-        read_grade = PersistentSubsectionGrade.read_grade(
-            user_id=self.params["user_id"],
-            usage_key=self.params["usage_key"],
-        )
-        self.assertEqual(read_grade.earned_all, 12)
-        self.assertEqual(read_grade.earned_graded, 8)
-
-    @ddt.data(True, False)
-    def test_save(self, already_created):
-        if already_created:
+        with patch('lms.djangoapps.grades.models.log') as log_mock:
+            with self.assertRaises(PersistentSubsectionGrade.DoesNotExist):
+                PersistentSubsectionGrade.update_grade(**self.params)
             PersistentSubsectionGrade.objects.create(**self.params)
+            self.params['earned_all'] = 12.0
+            self.params['earned_graded'] = 8.0
+            PersistentSubsectionGrade.update_grade(**self.params)
+            read_grade = PersistentSubsectionGrade.read_grade(
+                user_id=self.params["user_id"],
+                usage_key=self.params["usage_key"],
+            )
+            self.assertEqual(read_grade.earned_all, 12)
+            self.assertEqual(read_grade.earned_graded, 8)
+            log_mock.info.assert_called_with(
+                u"Persistent Grades: Persistent subsection grade updated: {0}".format(read_grade)
+            )
+
+    def test_save(self):
         module_prefix = "lms.djangoapps.grades.models.PersistentSubsectionGrade."
         with patch(
-            module_prefix + "objects.get_or_create",
+            module_prefix + "objects.update_or_create",
             wraps=PersistentSubsectionGrade.objects.get_or_create
-        ) as mock_get_or_create:
-            with patch(module_prefix + "update") as mock_update:
-                PersistentSubsectionGrade.save_grade(**self.params)
-                self.assertTrue(mock_get_or_create.called)
-                self.assertEqual(mock_update.called, already_created)
+        ) as mock_update_or_create:
+            PersistentSubsectionGrade.save_grade(**self.params)
+            self.assertTrue(mock_update_or_create.called)
+
+    def test_logging_for_save(self):
```

Comment:
Can the log check be done within the `test_save` method?
If you choose to keep it isolated in its own test method, then you should remove the call to `read_grade` here since it's out of context for what it's testing.